launcher/launcher/main.go

```diff
@@ -180,3 +190,59 @@ func startLauncher() error {
 
 	return r.Run(ctx)
 }
+
+// verifyFsAndMount checks the partitions/mounts are as expected, based on the command output reported by OS.
+func verifyFsAndMount() error {
+	// check protected_stateful_partition is encrypted and is on integrity protection
+	cryptsetupOutput, err := exec.Command("cryptsetup", "status", "/dev/mapper/protected_stateful_partition").Output()
+	if err != nil {
+		return err
+	}
+	matched := regexp.MustCompile(`type:\s+LUKS2`).FindString(string(cryptsetupOutput))
+	if len(matched) == 0 {
+		return fmt.Errorf("stateful partition is not LUKS2 formatted: \n%s", cryptsetupOutput)
+	}
+	matched = regexp.MustCompile(`integrity:\s+aead`).FindString(string(cryptsetupOutput))
```

Comment:
should also check aes-gcm with random IV: ` cipher:  aes-gcm-random`

Reply:
done